### PR TITLE
feat: uses sade instead of commander

### PR DESCRIPTION
**Pull request by maraisr**
Migrates to using sade instead of commander.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,12 +13,11 @@
     "compile": "tsc --project tsconfig.build.json",
     "set-permissions": "chmod +x ./dist/cli.js",
     "prepublish": "yarn build",
-    "cli:run": "node dist/cli.js",
-    "postbuild": "node dist/cli idea-exclude dist --lookabove"
```
Comment (maraisr) [inline package.json:17]: infinite loop when not running idea for this project 👀

File: src/cli.ts [hunk 2]
```diff
@@ -1,40 +1,46 @@
...
 program
-  .version('0.0.1')
-  .option('--lookabove', 'allows looking up the current folder in search of idea root')
-  .option('-i, --ignore <glob>', 'glob ignore mask')
-  .option('-r, --root <path>', 'root folder, defaults to process.cwd', process.cwd())
-  .arguments('<group> [glob...]')
-  .description('idea-exclude', {
-    group: 'an exclude group, for example all-node-modules',
-    glob: 'a selection glob'
-  })
```
Comment (maraisr) [inline src/cli.ts:16]: no direct translation into `sade` — but that is okay. Just extra things... the examples does a far better job at documenting this than anything else.

File: package.json (hunk 1 above)
Comment (theKashey) [reply package.json:17]: stop stop. Why?

Comment (maraisr) [reply package.json:17]: try it, run `yarn` in the this folder. After yarn, it builds and then looks like it runs it. The just `🔎 looking above /` a bajillion times.

Comment (maraisr) [reply package.json:17]: will add this also happens in mainline, not just this pr. Can revert this line.

Comment (theKashey) [reply package.json:17]: Probably that's a bug. It should auto-exclude own `dist` folder, and if it cannot - it's broken :)

Comment (maraisr) [reply package.json:17]: Perhaps we can fix this in another PR — ill remove this line from this one.

**Comment by theKashey** [comment]
cool beans. I know which tool I am gonna use next time 👍

**Pull request merged**